### ISSUE TITLE
Bug: M1 .app size is too big

### ISSUE THREAD

**Issue by mpogue2** [mpogue2/SquareDesk#627]
It's about 418MB, compare to X86: 180MB.
Most of that M1 Native app size is in QtWebEngineCore, which is 307MB (!).  That same file is only 89MB on the X86 version.

Maybe symbols are not stripped from the M1 versions?  Maybe I'm now supposed to do that manually myself at deploy time?
Maybe the macdeployqt script has some special option I'm supposed to be using now?

**Comment by mpogue2** [comment]
Ah.  The QtWebEngine Core is a Universal binary:

```
(base) mpogue@Mikes-MacBook-Pro Current % file QtWebEngineCore
QtWebEngineCore: Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit dynamically linked shared library x86_64] [arm64]
QtWebEngineCore (for architecture x86_64):	Mach-O 64-bit dynamically linked shared library x86_64
QtWebEngineCore (for architecture arm64):	Mach-O 64-bit dynamically linked shared library arm64
```

whereas on the X86-only version, it's not:

```
(base) mpogue@Mikes-MacBook-Pro Current % file QtWebEngineCore
QtWebEngineCore: Mach-O 64-bit dynamically linked shared library x86_64
```

**Comment by mpogue2** [comment]
The solution is this (at least for now, since I have separate M1 and X86 binaries):
https://eclecticlight.co/2020/07/30/instant-weight-loss-how-to-strip-universal-apps/

Basically, 
```
for each Qt* framework file f:
     mv f f_backup
     lipo -remove x86_64 f_backup -output f
     rm f_backup (assuming all is good)
```
QtWebEngineCore went from **308MB** to **145MB** (a savings of 163MB).

Of course, why is QtWebEngineCore in there right now anyway, given that the References tab is currently turned off?
Now I know this will all get fatter if/when I move to a single Universal Binary.  In the mean time, I think this .app file can be slimmed down quite a bit, with a little more work on the PackageIt_M1.command script.

**Comment by mpogue2** [comment]
An explanation of why there are so many frameworks included:
https://github.com/Homebrew/homebrew-core/issues/6161

otool -L on QtWebEngineCore shows that it depends on stuff like QtQml and QtQuick, which is surprising, because this app does not use QML at all.  To get rid of this, I'd have to do a custom build of Qt6, which is doable, but painful.

Here's where the space is going:

```
(base) mpogue@Mikes-MacBook-Pro Frameworks % find . -type f -name "Qt*" -exec ls -al {} \; | grep -v prl
-rwxr-xr-x  1 mpogue  admin  149152 Oct 21 04:05 ./QtWebEngineCore.framework/Versions/A/Helpers/QtWebEngineProcess.app/Contents/MacOS/QtWebEngineProcess
-rw-r--r--  1 mpogue  admin  387 Oct 21 03:01 ./QtWebEngineCore.framework/Versions/A/Helpers/QtWebEngineProcess.app/Contents/Resources/QtWebEngineProcess.entitlements
-rwxr-xr-x  1 mpogue  admin  144904400 Nov 26 19:14 ./QtWebEngineCore.framework/Versions/A/QtWebEngineCore
-rwxr-xr-x  1 mpogue  admin  353536 Nov 26 15:56 ./QtWebEngineWidgets.framework/Versions/A/QtWebEngineWidgets
-rwxr-xr-x  1 mpogue  admin  330912 Nov 26 15:56 ./QtQuickWidgets.framework/Versions/A/QtQuickWidgets
-rwxr-xr-x  1 mpogue  admin  1006112 Nov 26 15:56 ./QtPrintSupport.framework/Versions/A/QtPrintSupport
-rwxr-xr-x  1 mpogue  admin  15045640 Nov 26 15:56 ./QtGui.framework/Versions/A/QtGui
-rwxr-xr-x  1 mpogue  admin  1400504 Nov 26 15:56 ./QtDBus.framework/Versions/A/QtDBus
-rwxr-xr-x  1 mpogue  admin  270624 Nov 26 15:57 ./QtSerialPort.framework/Versions/A/QtSerialPort
-rwxr-xr-x  1 mpogue  admin  10598808 Nov 26 15:56 ./QtQuick.framework/Versions/A/QtQuick
-rwxr-xr-x  1 mpogue  admin  11106920 Nov 26 15:56 ./QtCore.framework/Versions/A/QtCore
-rwxr-xr-x  1 mpogue  admin  9109752 Nov 26 15:56 ./QtQml.framework/Versions/A/QtQml
-rwxr-xr-x  1 mpogue  admin  583920 Nov 26 15:56 ./QtWebChannel.framework/Versions/A/QtWebChannel
-rwxr-xr-x  1 mpogue  admin  1916992 Nov 26 15:56 ./QtMultimedia.framework/Versions/A/QtMultimedia
-rwxr-xr-x  1 mpogue  admin  1996480 Nov 26 15:57 ./QtVirtualKeyboard.framework/Versions/A/QtVirtualKeyboard
-rwxr-xr-x  1 mpogue  admin  1217752 Nov 26 15:56 ./QtOpenGL.framework/Versions/A/QtOpenGL
-rwxr-xr-x  1 mpogue  admin  12346104 Nov 26 15:56 ./QtWidgets.framework/Versions/A/QtWidgets
-rwxr-xr-x  1 mpogue  admin  1157264 Nov 26 15:56 ./QtPositioning.framework/Versions/A/QtPositioning
-rwxr-xr-x  1 mpogue  admin  2969496 Nov 26 15:56 ./QtNetwork.framework/Versions/A/QtNetwork
-rwxr-xr-x  1 mpogue  admin  740360 Nov 26 15:56 ./QtSvg.framework/Versions/A/QtSvg
-rwxr-xr-x  1 mpogue  admin  1245160 Nov 26 15:56 ./QtQmlModels.framework/Versions/A/QtQmlModels
-rwxr-xr-x  1 mpogue  admin  587832 Nov 26 15:56 ./QtSql.framework/Versions/A/QtSql
```

This one:

`-rwxr-xr-x  1 mpogue  admin  144904400 Nov 26 19:14 ./QtWebEngineCore.framework/Versions/A/QtWebEngineCore
`
was already "thinned" down to just the arm version.

**Comment by mpogue2** [comment]
Inside the PackageIt_M1.command file, I removed the x86_64 versions of the Qt libs from the Frameworks directory, since this is (right now) NOT a Universal binary.  Size of the SquareDesk_0.9.6.app file went down from 442Mb to 230Mb (more comparable to the X86 app, which is 173MB when installed).  

The size of the DMG file went down to 93MB (comparable to the X86 DMG which is 68MB).  I can live with this size for now.

**Comment by mpogue2** [comment]
Fixed by 9aebd85085ef866c4efacda702e4a4fa5020056c .
